clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -547,23 +549,33 @@ object GarbageCollector {
   private def getPreviousRunID(
       incrementalRun: Boolean,
       storageNS: String,
-      configMapper: ConfigMapper
+      configMapper: ConfigMapper,
+      iterations: Int
   ): String = {
     if (!incrementalRun) {
       return ""
     }
+    if (iterations < 1) {
+      throw RunIDException("Run ID iteration number cannot be smaller than 1")
+    }
     var previousRunID = ""
     val runIDsPath = new Path(s"${storageNS.stripSuffix("/")}/_lakefs/retention/gc/run_ids/")
     val runIDsFS = runIDsPath.getFileSystem(configMapper.configuration)
     val runIDs = runIDsFS.listFiles(runIDsPath, false)
-    if (runIDs.hasNext) {
-      previousRunID = runIDs.next().getPath.getName
-      println(s"----------------------- Using previous RUN ID: $previousRunID")
-    } else {
+    if (!runIDs.hasNext) {
       runIDsFS.close()
       throw RunIDException("No previous run ID")
     }
+    var runIDObject: LocatedFileStatus = null
+    for(_ <- 0 until iterations) {
+      if(!runIDs.hasNext) {
+        throw RunIDException("Required Run ID iteration doesn't exist")
+      }
+      runIDObject = runIDs.next()
+    }
```

Comment:
Consider

```scala
val runIDObject = try {
  runIDs.drop(iterations-1).next()
} finally {
  runIDsFS.close()
}
```
and possibly translating exceptions in a `catch` block before that `finally`.  As incentive, l. 572 should probably also close runIDsFS.

Reply:
[Convert it](https://github.com/treeverse/lakeFS/blob/ebb6049a3f04f3f18cb11ebb7bbf9ac4c36dd4cc/test/spark/app/src/main/scala/Sonnets.scala#L9-L26)?